shared/modules/node_module.go

```diff
@@ -0,0 +1,8 @@
+package modules
```

Comment:
The idea was something we could mock later on. since I see I have embedded `P2PAddressableModule` in there, I think I was thinking about P2P at the time of writing.

Reply:
Blocker:

We need comments here to describe this thing.

I agree with Olshansk

> I don't fully understand the logic behind this interface. Can you expand on the thought process? I think we might need to remove it.